src/modules/components/wallet/wallet.tsx

```diff
@@ -0,0 +1,58 @@
+import classNames from 'classnames';
+import { useEnsName } from 'wagmi';
+import { StateSkeletonBar } from '../../../core';
+import { type ICompositeAddress } from '../../types';
+import { addressUtils } from '../../utils';
+import { MemberAvatar } from '../member';
+
+export interface IWalletProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+    /**
+     * The connected user details.
+     */
+    user: ICompositeAddress;
+    /**
+     * Whether or not the app is connected to a wallet.
+     */
+    isConnected: boolean;
```

Comment:
If the user is not connected, we don't have an address to set. We can just remove the `isConnected` property, set the `user` property as optional and display the connect button / user avatar-ens depending if the `user` property is set